eclass/cargo.eclass

```diff
@@ -160,6 +180,37 @@ cargo_crate_uris() {
 		url="https://crates.io/api/v1/crates/${name}/${version}/download -> ${crate}.crate"
 		echo "${url}"
 	done
+
+	local git_crates_type
+	git_crates_type="$(declare -p GIT_CRATES)"
+	if [[ ${git_crates_type} == "declare -A "* ]]; then
+		local crate commit crate_uri crate_dir repo_ext feat_expr
+
+		for crate in "${!GIT_CRATES[@]}"; do
+			IFS=';' read -r crate_uri commit crate_dir <<< "${GIT_CRATES[${crate}]}"
+
+			case "${crate_uri}" in
+				https://github.com/*)
+					repo_ext=".gh"
```

Comment:
question: why repo ext is needed at all?

Reply:
Because pkgcheck complains if you have a tarball generated by GitHub without that.

Reply:
That's only for pypi packages and, even if have pypi in metadata, pkgcheck has an exemption when using a full hash (e.g. it *used* to warn for rapidyaml's test tarballs which has pypi in metadata but stopped after the exemption).

Reply:
Because pkgcheck complains if you have a tarball generated by GitHub without that.

Reply:
Same message?

I tried the eclass without .gh and there's no warning for me. The pkgcheck exception I mentioned works (aka it doesn't warn if full hash is in the name, do be sure your pkgcheck is up to date), and the extensions just seem useless. This isn't where .gh is needed, it's meant to avoid confusing with pypi tarballs, not crate snapshots with hash names.